src/v/iceberg/table_io.cc

```diff
@@ -40,9 +42,7 @@ ss::future<checked<size_t, metadata_io::errc>> table_io::upload_table_meta(
 ss::future<checked<int, metadata_io::errc>>
 table_io::download_version_hint(const version_hint_path& path) {
     return download_object<int>(path(), "iceberg::version_hint", [](iobuf b) {
-        iobuf_parser p(std::move(b));
-        auto sz = p.bytes_left();
-        auto version_str = p.read_string(sz);
+        auto version_str = b.linearize_to_string();
```

Comment:
are we just not concerned about utf8 encoding on external text we are parsing?

Reply:
The JSON parser validates this